includes/blocks/class-attributes.php

```diff
@@ -137,7 +137,10 @@ protected function boot() {
 		}
 
 		// @todo remove when implemented globally.
-		$attributes['data-block'] = $this->name;
+		if ( ! in_array( $this->area, [ 'view_block_primary', 'view_block_secondary' ] ) ) {
```

Comment:
Maybe it's better to include instead of excluding? It seems that the only area with re-rendering is the view_page_primary one. Another approach is to differentiate this somehow on JS side, but a quick hard-coding may be ok (with some "todo" though).